backend_new/src/services/application-flagged-set.service.ts

```diff
@@ -0,0 +1,951 @@
+import {
+  BadRequestException,
+  Inject,
+  Injectable,
+  Logger,
+  NotFoundException,
+  OnModuleInit,
+} from '@nestjs/common';
+import { SchedulerRegistry } from '@nestjs/schedule';
+import { CronJob } from 'cron';
+import dayjs from 'dayjs';
+import {
+  ApplicationReviewStatusEnum,
+  ApplicationStatusEnum,
+  FlaggedSetStatusEnum,
+  ListingsStatusEnum,
+  Prisma,
+  RuleEnum,
+} from '@prisma/client';
+import { PrismaService } from './prisma.service';
+import { mapTo } from '../utilities/mapTo';
+import { SuccessDTO } from '../dtos/shared/success.dto';
+import { ApplicationFlaggedSet } from '../dtos/application-flagged-sets/application-flagged-set.dto';
+import { PaginatedAfsDto } from '../dtos/application-flagged-sets/paginated-afs.dto';
+import { AfsQueryParams } from '../dtos/application-flagged-sets/afs-query-params.dto';
+import { AfsMeta } from '../dtos/application-flagged-sets/afs-meta.dto';
+import { OrderByEnum } from '../enums/shared/order-by-enum';
+import { View } from '../enums/application-flagged-sets/view';
+import { buildPaginationMetaInfo } from '../utilities/pagination-helpers';
+import { AfsResolve } from '../dtos/application-flagged-sets/afs-resolve.dto';
+import { User } from '../dtos/users/user.dto';
+import { Application } from '../dtos/applications/application.dto';
+import { IdDTO } from 'src/dtos/shared/id.dto';
+
+/*
+  this is the service for unit types
```

Comment:
is this comment accurate?

Reply:
my secrets have been revealed copy and pasting from other services lol

fixed!